app/Models/Doctor.php

```diff
@@ -4,8 +4,37 @@
 
 use Illuminate\Database\Eloquent\Factories\HasFactory;
 use Illuminate\Database\Eloquent\Model;
+use Illuminate\Database\Eloquent\SoftDeletes;
+use Illuminate\Foundation\Auth\User as Authenticatable;
+use Illuminate\Support\Facades\Hash;
+use Laravel\Passport\HasApiTokens;
 
-class Doctor extends Model
+class Doctor extends Authenticatable
 {
-    use HasFactory;
+    use HasFactory, HasApiTokens, SoftDeletes;
+
+    /**
+     * The attributes that are mass assignable.
+     *
+     * @var array
+    */
+    protected $fillable = [
+        'name', 'email', 'password', 'phone','specialization_id', 'gender'
+        , 'photo', 'degree_copy'
+    ];
+
+    public function setPasswordAttribute($value)
+    {
+        $this->attributes['password'] = Hash::make($value);
+    }
+
+    public function setPhotoAttribute($value)
+    {
+        $this->attributes['photo'] = $value->getClientOriginalName();
```

Comment:
Images should generate a unique name when it's uploaded

Reply:
have a question here to be discussed